pkg/option/config.go

```diff
@@ -3337,8 +3349,27 @@ func InitConfig(programName, configName string) func() {
 			} else {
 				// replace deprecated fields with new fields
 				ReplaceDeprecatedFields(m)
-				err := MergeConfig(m)
-				if err != nil {
+				var confArgs []string
+				for key := range m {
+					if key == "cluster-id" {
+						fmt.Println("cluster-id")
+					}
+					value := fmt.Sprintf("%v", m[key])
+					// The config flags may differ from the CLI flags
```

Comment:
What does this comment mean? I'm not sure I understand the difference between config and CLI flags.

Reply:
@romanspb80 I still think it would be worth detailing this comment as it may not be clear to folks reading the code that config flag != CLI flag. Maybe giving an example is enough?

Reply:
> @romanspb80 I still think it would be worth detailing this comment as it may not be clear to folks reading the code that config flag != CLI flag. Maybe giving an example is enough?

OK. Done:
https://github.com/cilium/cilium/pull/15490/commits/a1bfb0f5a0ca4043fc378c1bd21c81b3076e3c1f#diff-8a383aefc352ba8854dbab293a563fecf7f8916983d013ab36f677293996ca3cR2540-R2545

Reply:
Thanks!